source/parser.hera

```diff
@@ -350,14 +350,29 @@ UnaryExpression
     ws = insertTrimmingSpace(ws, "")
     return ["(", ...ws, exp, ")()"]
 
+UnaryWithoutParenthesizedAssignment
+  UnaryOp*:pre UnaryWithoutParenthesizedAssignmentBody:exp UnaryPostfix?:post ->
+    return processUnaryExpression(pre, exp, post)
+
+UnaryBody
+  ParenthesizedAssignment
+  UpdateExpression
+  ExpressionizedStatementWithTrailingCallExpressions
+  NestedNonAssignmentExtendedExpression
+
+UnaryWithoutParenthesizedAssignmentBody
+  UpdateExpression
+  ExpressionizedStatementWithTrailingCallExpressions
+  NestedNonAssignmentExtendedExpression
+
 UnaryPostfix
   QuestionMark
   ( __ ( As / Satisfies ) Type )+ -> { ts: true, children: $0 }
 
 # https://262.ecma-international.org/#prod-UpdateExpression
 UpdateExpression
   # NOTE: Not allowing whitespace betwen prefix and postfix increment operators and operand
-  UpdateExpressionSymbol UnaryExpression ->
+  UpdateExpressionSymbol UnaryWithoutParenthesizedAssignment ->
```

Comment:
Hmm, why does `++(x=5)` still work? Does it parse via another path?

Reply:
The `ParenthesizedAssignment` is for wrapping `x = 5` in implicit parentheses. An explicitly parenthesized assignment will be a `ParenthesizedExpression` containing an `AssignmentExpression`.